File: CPP/Day17.cpp
/*

Day 17: Program to find whether s is a substring of str or not using recursion.

*/


//Program
#include <bits/stdc++.h>
using namespace std;

bool isSubstring(string str, string s) {
    if (s.empty()) {
        return true;
    }
    if (str.empty()) {
        return false;
    }
    if (str.substr(0, s.size()) == s) {
        return true;
    }
    return isSubstring(str.substr(1), s);
}

int main() {
    string str;
    string s;

    cout << "Enter the string: ";
    getline(cin, str);
    cout << "Enter the substring: ";
    cin >> s;
    
    if (isSubstring(str, s)) {
        cout << s << " is a substring of: " << str << endl;
    } else {
        cout << s << " is not a substring of: " << str << endl;
    }
    return 0;
}





/*

Input and Output:

Enter the string: This is IODump Chaitra Challenge
Enter the substring: IODump
IODump is a substring of: This is IODump Chaitra Challenge

*/

File: Python/Day17.py
"""

Day 17:	Program to find whether s is a substring of str or not using recursion.

"""

#Program
def isSubstring(s2, s1):
	if s2 in s1:
		return s1.index(s2)
	return -1



s1 = input("Enter the string: ")
s2 = input("Enter the substring: ")

result = isSubstring(s2, s1)
if result == -1:
	print(f"\"{s2}\" is NOT a SUBSTRING of \"{s1}\" ")
else:
	print(f"\"{s2}\" is a SUBSTRING of \"{s1}\" ")



"""

Input and Output: 

Enter the string: Welcome to IODump Chaitra Challenge
Enter the substring: Challenge
"Challenge" is a SUBSTRING of "Welcome to IODump Chaitra Challenge"

"""